icu4c/source/test/testdata/BRULES.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (C) 2018 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+
+from distutils.sysconfig import parse_makefile
+
+from buildtool import *
+
+
+def generate(config, glob, common_vars):
+    build_dirs = ["{OUT_DIR}", "{TMP_DIR}"]
+
+    requests = []
+    requests += generate_rb(config, glob, common_vars)
+    requests += generate_sprep(config, glob, common_vars)
+    requests += generate_conv(config, glob, common_vars)
+    requests += generate_other(config, glob, common_vars)
+    requests += generate_copy(config, glob, common_vars)
+
+    all_output_files = list(sorted(utils.get_all_output_files(requests)))
+    all_output_files_with_tmp = list(sorted(utils.get_all_output_files(requests, include_tmp=True)))
+    testdata_list_file = TmpFile("testdata.lst")
+    requests += [
+        PrintFileRequest(
+            name = "testdata_list",
+            output_file = testdata_list_file,
+            content = "\n".join(file.filename for file in all_output_files)
+        ),
+        VariableRequest(
+            name = "testdata_all_output_files",
+            input_files = all_output_files_with_tmp + [testdata_list_file]
+        )
+    ]
+
+    return (build_dirs, requests)
+
+
+def generate_rb(config, glob, common_vars):
+    mk_vars = parse_makefile("{GLOB_DIR}/tstfiles.mk".format(**common_vars))
+    basenames = [v[:-4] for v in mk_vars["TEST_RES_SOURCE"].split()]
+    basenames += [
+        "casing",
```

Comment:
these should be in an external file. If you're already reading tstfiles.mk ,couldn't the list come from there?

Reply:
`tstfiles.mk` contains only a very small subset of these file names.  Previously the list was hard-coded into testdata.mak and Makefile.in.  It is deleted now from those two and put here.